sdk/storage/Azure.Storage.DataMovement.Blobs/src/BlobDirectoryStorageResourceContainerOptions.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Storage.DataMovement;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.DataMovement.Models;
+
+namespace Azure.Storage.DataMovement.Blobs
+{
+    /// <summary>
+    /// Options bag for <see cref="BlobDirectoryStorageResourceContainer"/>
+    /// </summary>
+    public class BlobDirectoryStorageResourceContainerOptions
+    {
+        /// <summary>
+        /// Optional. The <see cref="BlobTraits"/> for when calling the
+        /// <see cref="BlobDirectoryStorageResourceContainer.GetStorageResourcesAsync(System.Threading.CancellationToken)"/>.
+        /// </summary>
+        public BlobTraits Traits { get; set; }
+
+        /// <summary>
+        /// Optional. The <see cref="BlobStates"/> for when calling the
+        /// <see cref="BlobDirectoryStorageResourceContainer.GetStorageResourcesAsync(System.Threading.CancellationToken)"/>.
+        /// </summary>
+        public BlobStates States { get; set; }
+
+        /// <summary>
+        /// When calling <see cref="BlockBlobStorageResource.CopyBlockFromUriAsync(StorageResource, HttpRange, bool, long, StorageResourceCopyFromUriOptions, System.Threading.CancellationToken)"/>
+        /// </summary>
+        public BlockBlobStorageResourceServiceCopyOptions CopyOptions { get; set; }
```

Comment:
Consider changing this to not just be for block blob options. Related to https://github.com/Azure/azure-sdk-for-net/issues/32728